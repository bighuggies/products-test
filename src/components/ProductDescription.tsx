import * as React from 'react';
import styled from 'styled-components';

interface IProps {
  id: string;
  description: string;
}

const StyledProductDescription = styled.p`
  font-style: italic;
  margin: 0 0 1rem 0;
`;

class ProductDescription extends React.PureComponent<IProps> {
  public render() {
    return (
      <StyledProductDescription id={this.props.id}>
        {this.props.description}
      </StyledProductDescription>
    );
  }
}

export default ProductDescription;
