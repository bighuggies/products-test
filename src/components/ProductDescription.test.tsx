import { mount } from 'enzyme';
import * as React from 'react';

import ProductDescription from './ProductDescription';

test('renders without crashing', () => {
  const mounted = mount(
    <ProductDescription id={'id1'} description={'hello world'} />
  );

  expect(mounted.debug()).toMatchSnapshot();
});
