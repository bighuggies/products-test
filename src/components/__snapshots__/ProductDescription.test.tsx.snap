// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<ProductDescription id=\\"id1\\" description=\\"hello world\\">
  <styled.p id=\\"id1\\">
    <p className=\\"sc-bdVaJa hjusWQ\\" id=\\"id1\\">
      hello world
    </p>
  </styled.p>
</ProductDescription>"
`;
